src/DIRAC/Core/Utilities/TimeUtilities.py

```diff
@@ -73,32 +75,45 @@ def timed(*args, **kw):
                     except IndexError:
                         argsLen = ""
 
-        print("%s Exec time ===> function %r %s -> %2.2f sec" % (pre, method.__name__, argsLen, te - ts))
+        gLogger.info("%s Exec time ===> function %r %s -> %2.2f sec" % (pre, method.__name__, argsLen, te - ts))
         return result
 
     return timed
 
 
 def toEpoch(dateTimeObject=None):
     """
-    Get seconds since epoch
+    Get seconds since epoch. Accepts datetime or date objects
     """
-    if not dateTimeObject:
-        dateTimeObject = datetime.datetime.utcnow()
-    return calendar.timegm(dateTimeObject.utctimetuple())
+    return toEpochMilliSeconds(dateTimeObject) // 1000
 
 
 def toEpochMilliSeconds(dateTimeObject=None):
     """
     Get milliseconds since epoch
     """
-    return toEpoch(dateTimeObject) * 1000
+    if dateTimeObject == None:
+        dateTimeObject = datetime.datetime.utcnow()
+    if dateTimeObject.resolution == datetime.timedelta(days=1):
```

Comment:
What is this for? In case it's a `date`?

Reply:
Yes exactly